### PR TITLE
note service 

### PR DESCRIPTION
**Pull request by zakaria-root**
the crud of note element & module

### CODE REVIEW

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteElementCreationRequest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+public record NoteElementCreationRequest(
+    @NotBlank
+    String NoteElementId,
+    @Min(0)
+    @Max(20)
+    @NotNull
+    float note,
+    @NotNull
+    boolean redoublant,
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteElementCreationRequest.java:17]: default to false

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteModuleController.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        NoteModule foundModule = noteModuleService.findById(noteModuleId);
+        ResponseEntity<ModuleResponse> moduleresponse =  moduleServiceFeignClient.getModuleByCode(foundModule.getCodeModule());
+        NoteModuleWithModuleResponse foundModuleResponse = noteModuleMapper.toNoteModuleWithModuleResponse(foundModule);
+        foundModuleResponse = foundModuleResponse.setElementResponse(moduleresponse.getBody());
+        return ResponseEntity
+                .ok()
+                .body(foundModuleResponse);
+    }
+//
+    @GetMapping
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteModuleController.java:62]: are there any needs of getAllNotes ?
i think it should be given an id of session and returns a list of its notes

File: note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        NoteElement foundNoteElement = noteElementService.findById(noteElementId);
+        ResponseEntity<ElementResponse> elementResponse =  elementServiceFeignClient.getElementByCode(foundNoteElement.getCodeElement());
+        NoteElementWithElementResponse foundModuleResponse = noteElementMapper.toNoteElementWithElementResponse(foundNoteElement);
+        foundModuleResponse = foundModuleResponse.setElementResponse(elementResponse.getBody());
+        return ResponseEntity
+                .ok()
+                .body(foundModuleResponse);
+    }
+    @GetMapping
+    public  ResponseEntity<NoteElementPagingResponse> getAll(@RequestParam(defaultValue = "0") @Min(0) int page,
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java:64]: checkout NoteModuleController

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteElementCreationRequest.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package ma.enset.noteservice.dto;
+
+import jakarta.validation.constraints.*;
+import jakarta.ws.rs.DefaultValue;
+import net.bytebuddy.implementation.bind.annotation.Default;
+
+import java.math.BigDecimal;
+
+public record NoteElementCreationRequest(
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteElementCreationRequest.java:9]: codeSession ??
checkout class diagram.

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteElementPagingResponse.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package ma.enset.noteservice.dto;
+
+import lombok.Builder;
+
+import java.util.List;
+
+@Builder
+public record NoteElementPagingResponse(
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteElementPagingResponse.java:8]: there is no need of this (no need of getAllNotes)

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteElementWithElementResponse.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+public record NoteElementWithElementResponse(
+    String noteElementId,
+    float note,
+    boolean redoublant,
+
+    String codeElement,
+    ElementResponse element
+) {
+
+    public NoteElementWithElementResponse setElementResponse(ElementResponse element) {
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteElementWithElementResponse.java:13]: You can use mapper

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteModuleCreationRequest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    @Max(20)
+    @NotNull
+    float note,
+
+    @NotNull
+    Resultat resultat,
+    @NotNull
+    boolean redoublant,
+    @NotNull
+    boolean rattrapage,
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteModuleCreationRequest.java:26]: default(false)

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteModuleCreationRequest.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    String NoteModuleId,
+    @Min(0)
+    @Max(20)
+    @NotNull
+    float note,
+
+    @NotNull
+    Resultat resultat,
+    @NotNull
+    boolean redoublant,
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteModuleCreationRequest.java:24]: default(false)

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteModulePagingResponse.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package ma.enset.noteservice.dto;
+
+import lombok.Builder;
+
+import java.util.List;
+
+@Builder
+public record NoteModulePagingResponse(
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteModulePagingResponse.java:8]: Checkout NoteElementPagingResponse

File: note-service/src/main/java/ma/enset/noteservice/dto/NoteModuleResponse.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
+package ma.enset.noteservice.dto;
+
+import ma.enset.noteservice.enums.Resultat;
+
+import javax.xml.transform.Result;
+
+public record NoteModuleResponse(
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/dto/NoteModuleResponse.java:7]: Same as NoteElementResponse

File: note-service/src/main/java/ma/enset/noteservice/model/NoteElement.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.NoArgsConstructor;
+
+import java.math.BigDecimal;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Data
+public class NoteElement {
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/model/NoteElement.java:18]: codeSession ??
checkout class diagram

File: note-service/src/main/java/ma/enset/noteservice/model/NoteModule.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ma.enset.noteservice.enums.Resultat;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Data
+public class NoteModule {
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/model/NoteModule.java:15]: codeSession ??
checkout class diagram

File: note-service/src/main/java/ma/enset/noteservice/service/NoteElementService.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+import ma.enset.noteservice.exception.ElementNotFoundException;
+import ma.enset.noteservice.exception.InternalErrorException;
+import ma.enset.noteservice.model.NoteElement;
+import ma.enset.noteservice.model.NoteModule;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import java.util.List;
+
+public interface NoteElementService {
```
Comment (Elbouchouki) [inline note-service/src/main/java/ma/enset/noteservice/service/NoteElementService.java:13]: Must have updateMany

**Review by Elbouchouki — CHANGES_REQUESTED** [review]
try to make it work without connecting it to other services.

File: note-service/src/main/java/ma/enset/noteservice/service/NoteElementService.java (hunk 13 above)
Comment (zakaria-root) [reply note-service/src/main/java/ma/enset/noteservice/service/NoteElementService.java:13]: Why do I need to update many note at the same time ?

Comment (Elbouchouki) [reply note-service/src/main/java/ma/enset/noteservice/service/NoteElementService.java:13]: From service to service, when updating multiple objects (notes), we can do it the way you see it, (for every object we send an update request), but it is very slow and inefficient. what we can do is update all objects with only one call. we simply send all objects we want to update in the body request and update them all. (checkout utilisateur-service or element-service)

**Pull request closed without merging**